pkg/controllers/cloud/configmap_controller.go

```diff
@@ -59,38 +87,81 @@ func (r *ConfigMapReconciler) Reconcile(_ context.Context, req ctrl.Request) (ct
 		r.Log.Error(err, "exiting nephe-controller")
 		os.Exit(1)
 	}
+
+	if err := r.setCloudSyncInterval(newControllerConfig.CloudSyncInterval); err != nil {
+		// Unable to continue as interval is not valid.
+		r.Log.Error(err, "exiting nephe-controller")
+		os.Exit(1)
+	}
+
 	return ctrl.Result{}, nil
 }
 
 // setCloudResourcePrefix sets the CloudResourcePrefix, returns error when invalid.
 func (r *ConfigMapReconciler) setCloudResourcePrefix(prefix string) error {
 	if len(prefix) == 0 {
-		r.Log.Info("Setting CloudResourcePrefix to default", "Old", r.ControllerConfig.CloudResourcePrefix,
-			"New", config.DefaultCloudResourcePrefix)
+		r.Log.Info("Setting CloudResourcePrefix to default", "CloudResourcePrefix", config.DefaultCloudResourcePrefix)
 		r.ControllerConfig.CloudResourcePrefix = config.DefaultCloudResourcePrefix
 		return nil
 	}
 
 	if !(config.ValidateName(prefix)) {
 		return fmt.Errorf("invalid CloudResourcePrefix %s, "+
-			"First and last characters should be alphanumeric and "+
-			"'-' characters are allowed only in the middle", prefix)
+			"Only alphanumeric and '-' characters are allowed."+
+			" Special character '-' is only allowed at the middle", prefix)
 	}
 
 	// Update the CloudResourcePrefix
 	if r.ControllerConfig.CloudResourcePrefix != prefix {
-		r.Log.V(1).Info("Updating the CloudResourcePrefix",
-			"Old", r.ControllerConfig.CloudResourcePrefix, "New", prefix)
+		r.Log.V(1).Info("Updating the CloudResourcePrefix", "CloudResourcePrefix", prefix)
 		r.ControllerConfig.CloudResourcePrefix = prefix
 	}
 	return nil
 }
 
+// setCloudSyncInterval sets the CloudSyncInterval, returns error when invalid.
+func (r *ConfigMapReconciler) setCloudSyncInterval(interval int64) error {
+	if interval == 0 {
+		r.Log.Info("Setting CloudSyncInterval to default", "CloudSyncInterval", config.DefaultCloudSyncInterval)
+		r.ControllerConfig.CloudSyncInterval = config.DefaultCloudSyncInterval
+		r.invokeConfigMapHandler(controllerConfigTypeInterval)
+		return nil
+	}
+
+	if interval < config.DefaultCloudSyncInterval {
+		r.ControllerConfig.CloudSyncInterval = config.DefaultCloudSyncInterval
+		r.invokeConfigMapHandler(controllerConfigTypeInterval)
+		r.Log.Error(nil, "invalid CloudSyncInterval %d, "+
+			"CloudSyncInterval should be >= %d seconds. So using default interval", interval, config.DefaultCloudSyncInterval)
+		return nil
+	}
+
+	// Update the CloudSyncInterval
+	if r.ControllerConfig.CloudSyncInterval != interval {
+		r.Log.V(1).Info("Updating the CloudSyncInterval", "CloudSyncInterval", interval)
+		r.ControllerConfig.CloudSyncInterval = interval
+		r.invokeConfigMapHandler(controllerConfigTypeInterval)
+	}
+	return nil
+}
+
+func (r *ConfigMapReconciler) registerConfigMapHandler(handler ConfigMapHandler, types controllerConfigType) {
+	configmapHandlerFunc[types] = append(configmapHandlerFunc[types], handler)
+}
+
+func (r *ConfigMapReconciler) invokeConfigMapHandler(types controllerConfigType) {
```

Comment:
I think its type, not types

Reply:
type is a keyword. Thats why used types.

Reply:
Change it to configType

Reply:
Solved.